lib/data/external/AwsClient.js

```diff
@@ -403,21 +405,28 @@ class AwsClient {
     callback) {
         const destBucketName = request.bucketName;
         const destObjectKey = request.objectKey;
+        const headers = request.headers;
         const destAwsKey = this._createAwsKey(destBucketName, destObjectKey,
           this._bucketMatch);
 
         const sourceAwsBucketName =
           config.getAwsBucketName(sourceLocationConstraintName);
 
-        const metadataDirective = request.headers['x-amz-metadata-directive'];
-        const metaHeaders = trimXMetaPrefix(getMetaHeaders(request.headers));
-        this._client.copyObject({
+        const metadataDirective = headers['x-amz-metadata-directive'];
+        const metaHeaders = trimXMetaPrefix(getMetaHeaders(headers));
+        const s3Params = {
             Bucket: this._awsBucketName,
             Key: destAwsKey,
             CopySource: `${sourceAwsBucketName}/${sourceKey}`,
             Metadata: metaHeaders,
             MetadataDirective: metadataDirective,
-        }, err => {
+        };
+        const isAWSServerSideEncrytion = headers[locationHeader] ?
```

Comment:
what if the destination bucket is an AWS serviceSideEncryption location?

Reply:
We only care about the object metadata location.
Each objects has its own location and we use this location to encrypt it or not.